### PR TITLE
fix(iOS): `connectToProtectedSSID` hangs indefinitely

### PR DESCRIPTION
**Pull request by thomasttvo**
https://github.com/JuanSeBestia/react-native-wifi-reborn/pull/397/files?w=1 attempted a fix for https://github.com/JuanSeBestia/react-native-wifi-reborn/issues/379. This PR marks the `timeout` parameter as `nonnull` so the red alert box no longer shows. The PR fails to change the TS types to reflect the nonnull-ness of the timeout parameter, so if you don't pass a timeout, RN doesn't like it and won't call the native method. However, if you look closely this `timeout` parameter isn't used by both iOS and Android, so it seems the right change here is to remove it. 

That said, the root cause of this is us getting confused by having 2 methods `connectToProtectedWifiSSID` and `connectToProtectedSSID` doing the same thing with the sole difference being whether the parameter is in the form of an object or not. This led me to thinking maybe we should just keep one of them, and this is what this PR does.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,7 +149,7 @@ The plugin provides props for extra customization. Every time you change the pro
 ```javascript
 import WifiManager from "react-native-wifi-reborn";
 
-WifiManager.connectToProtectedWifiSSID(ssid, password, isWep).then(
```
Comment (thomasttvo) [inline README.md:152]: this example here even hints that `connectToProtectedWifiSSID` is the one we should use. However, the example is incorrect in that the parameters need to be an object.

**Comment by shancock884** [comment]
If I understand correctly, this PR is also related to issue #413, where it is pointed out that the native connectToProtectedSSID function differs between IOS (5 args) and Android (4 args).

The issue with removing the function completely from the native code, is that it will break anyone's code who is already using the old function.
An idea could be to also add a JS wrapper to index.js, like so:
```javascript
WifiManager.connectToProtectedSSID = async (SSID, password = null, isWEP = false, isHidden = false, timeout = 15) => {
    return await WifiManager.connectToProtectedWifiSSID({ SSID, password, isWEP, isHidden, timeout });
}
```
This would then allow anyone using the old function prototype with either 4 or 5 arguments to keep doing so.

**Comment by DavideViolante** [comment]
The confusion is caused mainly by the following PR that introduced and mixed another method: https://github.com/JuanSeBestia/react-native-wifi-reborn/pull/355/

**Comment by DavideViolante** [comment]
Removing `connectToProtectedSSID` would be a breaking change and actually that method is the "main" one. The new one that introduced the timeout param is `connectToProtectedWifiSSID` and accept an object as param. I agree that this is very confusing and there should be only one method for connecting to a ssid.

I suggest a PR similar to what @shancock884 said above, merging the 2 methods into one.

**Pull request closed without merging**